File: backend/src/main/java/outcobra/server/model/MarkValue.java
package outcobra.server.model;

import javax.persistence.Entity;
import javax.validation.constraints.NotNull;

import lombok.*;

@Data
@Entity
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class MarkValue extends Mark {
    @NotNull
    private Double value;

    public MarkValue(Double value, Double weight) {
        this.value = value;
        this.weight = weight;
    }

    @Override
    public double getValue() {
        return value;
    }

    @Override
    public double getWeight() {
        return weight;
    }
}

File: backend/src/main/kotlin/outcobra/server/model/MarkValue.kt
package outcobra.server.model

import outcobra.server.model.interfaces.ParentLinked
import javax.persistence.Entity
import javax.persistence.OneToOne
import javax.validation.constraints.NotNull

@Entity
class MarkValue(@NotNull private var value: Double = 0.0,
                weight: Double = 1.0,
                markGroup: MarkGroup? = null,
                description: String = "",
                @OneToOne(mappedBy = "mark") var exam: Exam? = null) : Mark(weight, description, markGroup) {

    override fun getValue(): Double {
        return value
    }

    override val parent: ParentLinked?
        get() = markGroup

}
